pkg/platform/abstract/platform.go

```diff
@@ -1106,3 +1150,23 @@ func (ap *Platform) getBaseImagesOverrides() map[string]string {
 func (ap *Platform) getOnbuildImagesOverrides() map[string]string {
 	return ap.Config.ImageRegistryOverrides.OnbuildImageRegistries
 }
+
+func (ap *Platform) transformDisplayNameToName(projectConfig *platform.ProjectConfig) {
+	ap.Logger.WarnWith("Transforming display name",
+		"displayName", projectConfig.Spec.DisplayName,
+		"name", projectConfig.Meta.Name)
+
+	// trim spaces
+	displayName := strings.TrimSpace(projectConfig.Spec.DisplayName)
+
+	// lower case
+	displayName = strings.ToLower(displayName)
+
+	// no spaces
+	displayName = strings.ReplaceAll(displayName, " ", "-")
+
+	// no underscores
+	displayName = strings.ReplaceAll(displayName, "_", "-")
```

Comment:
import + use `inflect.Dasherize`

Reply:
because it is not `go get` able, I used `flect` instead